rasa/cli/data.py

```diff
@@ -472,7 +472,7 @@ def _migrate_model_config(args: argparse.Namespace) -> None:
     if fallback_rule:
         new_rules.append(fallback_rule)
 
-    if new_rules:
+    if new_rules or model_configuration["policies"]:
```

Comment:
So the behaviour we want is to always create a rules file even if empty?

Reply:
I am not sure what is the best approach so what I do now is creating the `rules.yml` file even if empty. However, after reading the warning above I think I should not create an empty `rules.yml` file. Also, I should change the implementation to the following :
- If there are no rules, then the `rules.yml` file should not be created.
- Adjust the success message to say something like "_The migration generated 0 rules so the file 'data/rules.yml' was not created._"
- The migration should edit the `config.yml` file by removing the MappingPolicy but since there are no rules, it should not append the RulePolicy.
So, I should actually make the opposite changes from the ones that I have now. What is happening right now is described in the description. Maybe @koaning has a suggestion on this one? Thank you!

Reply:
I don't see the harm in creating an empty `rules.yml` file or a `RulePolicy` that doesn't influence the conversation. But this is a *highly personal opinion that is created by thinking outloud*. One thing that I do not understand though, is `rasa data convert config` a new command? I couldn't find it on the docs.

Reply:
So, looking at the codebase : 
- it will raise a warning [here](https://github.com/RasaHQ/rasa/blob/b13e70bbbb8b774fb66340af96914da24d244465/rasa/core/policies/ensemble.py#L168) if there is a `RulePolicy` in the pipeline but no rules available so I guess until now the best practice was to **not** have an empty `rules.yml` file. 
- Also, this [test](https://github.com/RasaHQ/rasa/blob/b13e70bbbb8b774fb66340af96914da24d244465/tests/core/test_ensemble.py#L471) is there to check that the warning is raised correctly.

The `rasa data convert config` command is mentioned in the docs [here](https://rasa.com/docs/rasa/migration-guide) where it also says 
> _It will also add a rules.yml **_if necessary_**._

So if it is not necessary (eg. no rules available), it will not be created.

So, it looks like everything is aligned to not having the empty `rules.yml` file but of course the current state of the code is not necessarily the best practice and that it should not be questioned. I agree that it is no harm at all to have an empty rules.yml file or a RulePolicy that is not affecting the conversation. 

However, correct me if I am wrong but I think the goal of a migration is to convert what the user **already** has to the corresponding format of the new Rasa version and not what might be useful for the user to have. So based on that I would stick with what the current logic is and not create an empty rules file nor having a `RulePolicy` if there are no rules available. So in this PR I am in favour of : 
- only adjusting the success message to reflect that the `rules.yml` is not created and 
- make sure that the MappingPolicy is removed from the pipeline.
What do you think ?

Reply:
I'm less comfortable with having a strong opinion here because this is an area of the codebase that I'm unfamiliar with. That said, I might prefer having a `rules.yml` without any rules in it. The reason is that practically, a user will use rules somewhere in their assistant at some point. Might as well have a file/settings around that's easy to add rules to.

Reply:
@koaning I agree with the logic you proposed since from a user's perspective is more convenient to already have a `rules.yml` file even if empty. Maybe it makes also a better user experience because if the file is not there and they need rules, they might not be sure if they should create it and where exactly (so they will have to look in the docs or in the examples) or they might feel they have something wrong in their setup if they don't see it there, etc.